python/tvm/script/node.py

```diff
@@ -114,6 +114,11 @@ def check_index(index: Union[int, PrimExpr]):
                 check_index(index.start)
                 check_index(index.stop)
                 slices.append(index)
+            elif isinstance(index, BufferSlice):
+                for s in index.slices:
+                    if isinstance(s, Slice):
+                        check_index(s.start)
+                slices.append(index)
```

Comment:
I don't think this is correct. I believe you should should assume that the BufferSlice already checked that its indices were correct. Instead you want to check if the buffer dtype is correct (int32).

I am also a little confused as to why this case is not handled by the PrimExpr case?

Reply:
Python does not have type conversion system. The only thing we do is converting `BufferSlice` to `BufferLoad` when creating TVM Object. 

In this case, it still in the Python system, so can not be converted into `BufferLoad`